src/blob/core.rs

```diff
@@ -198,12 +198,12 @@ where
         Ok(blob)
     }
 
-    async fn raw_records(&self) -> Result<RawRecords> {
+    async fn raw_records(&self, validate_data_during_index_regen: bool) -> Result<RawRecords> {
```

Comment:
Here it is better to name the argument `validate_data`, because `raw_records` is not directly related to regenerating indexes